File: upload-images.py
#!/usr/bin/env python3
"""
سكريبت لتحميل الصور إلى المنتجات في Supabase
Script to upload images to products in Supabase
"""

import json
import os
import sys
from pathlib import Path

# إضافة المسار للوحدات
sys.path.insert(0, str(Path(__file__).parent))

try:
    from supabase import create_client, Client
except ImportError:
    print("❌ طلب التثبيت: pip install supabase")
    print("❌ Install required: pip install supabase")
    sys.exit(1)


def load_env():
    """تحميل متغيرات البيئة من .env.local"""
    env_path = Path(__file__).parent / ".env.local"
    env_vars = {}
    
    if env_path.exists():
        with open(env_path, 'r', encoding='utf-8') as f:
            for line in f:
                line = line.strip()
                if '=' in line and not line.startswith('#'):
                    key, value = line.split('=', 1)
                    env_vars[key.strip()] = value.strip()
    
    return env_vars


def update_products_images():
    """تحديث صور المنتجات في Supabase"""
    
    # تحميل البيانات
    env = load_env()
    supabase_url = env.get("VITE_SUPABASE_URL")
    supabase_key = env.get("VITE_SUPABASE_ANON_KEY")
    
    if not supabase_url or not supabase_key:
        print("❌ خطأ: لم يتم العثور على بيانات Supabase في .env.local")
        print("❌ Error: Supabase credentials not found in .env.local")
        return False
    
    # إنشاء اتصال Supabase
    supabase = create_client(supabase_url, supabase_key)
    
    # قراءة بيانات الصور
    products_file = Path(__file__).parent / "products-with-images.json"
    
    if not products_file.exists():
        print(f"❌ لم يتم العثور على الملف: {products_file}")
        return False
    
    with open(products_file, 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    print("🔧 تحديث الصور في Supabase...\n")
    
    # تحديث الفئات
    print("📂 تحديث صور الفئات...")
    for category in data.get("categories", []):
        try:
            response = supabase.table("categories").update({
                "image_url": category["image_url"]
            }).eq("name_en", category["name_en"]).execute()
            print(f"  ✅ {category['name_en']}")
        except Exception as e:
            print(f"  ❌ {category['name_en']}: {str(e)}")
    
    # تحديث المنتجات
    print("\n📦 تحديث صور المنتجات...")
    for product in data.get("products", []):
        try:
            response = supabase.table("products").update({
                "image_url": product["image_url"]
            }).eq("name_en", product["name_en"]).execute()
            print(f"  ✅ {product['name_en']}")
        except Exception as e:
            print(f"  ❌ {product['name_en']}: {str(e)}")
    
    # تحديث العروضات
    print("\n🎁 تحديث صور العروضات...")
    for offer in data.get("offers", []):
        try:
            response = supabase.table("offers").update({
                "image_url": offer["image_url"]
            }).eq("title_en", offer["title_en"]).execute()
            print(f"  ✅ {offer['title_en']}")
        except Exception as e:
            print(f"  ❌ {offer['title_en']}: {str(e)}")
    
    print("\n✅ تم تحديث جميع الصور بنجاح!")
    print("✅ All images updated successfully!")
    return True


if __name__ == "__main__":
    success = update_products_images()
    sys.exit(0 if success else 1)

File: upload-images.js
#!/usr/bin/env node

import { createClient } from "@supabase/supabase-js";
import * as fs from "fs";
import * as path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load .env.local
function loadEnv() {
  const envPath = path.join(__dirname, ".env.local");
  const env = {};

  if (fs.existsSync(envPath)) {
    const content = fs.readFileSync(envPath, "utf-8");
    content.split("\n").forEach((line) => {
      const [key, value] = line.split("=");
      if (key && value) {
        env[key.trim()] = value.trim();
      }
    });
  }

  return env;
}

async function uploadImages() {
  try {
    const env = loadEnv();
    const supabaseUrl = env.VITE_SUPABASE_URL;
    const supabaseKey = env.VITE_SUPABASE_ANON_KEY;

    if (!supabaseUrl || !supabaseKey) {
      console.error("❌ Error: Supabase credentials not found in .env.local");
      process.exit(1);
    }

    const supabase = createClient(supabaseUrl, supabaseKey);

    // Read products data
    const dataPath = path.join(__dirname, "products-with-images.json");
    if (!fs.existsSync(dataPath)) {
      console.error(`❌ File not found: ${dataPath}`);
      process.exit(1);
    }

    const data = JSON.parse(fs.readFileSync(dataPath, "utf-8"));

    console.log("🔧 Uploading images to Supabase...\n");

    // Update categories
    console.log("📂 Updating category images...");
    for (const category of data.categories) {
      try {
        const { error } = await supabase
          .from("categories")
          .update({ image_url: category.image_url })
          .eq("name_en", category.name_en);

        if (error) {
          console.log(`  ⚠️ ${category.name_en}: ${error.message}`);
        } else {
          console.log(`  ✅ ${category.name_en}`);
        }
      } catch (e) {
        console.log(`  ❌ ${category.name_en}: ${e.message}`);
      }
    }

    // Update products
    console.log("\n📦 Updating product images...");
    for (const product of data.products) {
      try {
        const { error } = await supabase
          .from("products")
          .update({ image_url: product.image_url })
          .eq("name_en", product.name_en);

        if (error) {
          console.log(`  ⚠️ ${product.name_en}: ${error.message}`);
        } else {
          console.log(`  ✅ ${product.name_en}`);
        }
      } catch (e) {
        console.log(`  ❌ ${product.name_en}: ${e.message}`);
      }
    }

    // Update offers
    console.log("\n🎁 Updating offer images...");
    for (const offer of data.offers) {
      try {
        const { error } = await supabase
          .from("offers")
          .update({ image_url: offer.image_url })
          .eq("title_en", offer.title_en);

        if (error) {
          console.log(`  ⚠️ ${offer.title_en}: ${error.message}`);
        } else {
          console.log(`  ✅ ${offer.title_en}`);
        }
      } catch (e) {
        console.log(`  ❌ ${offer.title_en}: ${e.message}`);
      }
    }

    console.log("\n✅ All images updated successfully!");
    console.log("🔄 Refresh your browser to see the changes.");
  } catch (error) {
    console.error("❌ Error:", error.message);
    process.exit(1);
  }
}

uploadImages();
